### PR TITLE
P0.2 · [RCM1] First contact establishes a gate row, so silence #1 stops being permanent

### LINKED ISSUE #278: P0.2 · [RCM1] First contact establishes a gate row, so silence #1 stops being permanent
Part of #257.

## User story

As someone who installed the app and turned notifications on, I want the system to know I exist — before I have done anything worth signalling.

## The mechanism, exactly

The waker's whole query is `SELECT subject_id FROM engagement_gate WHERE eligible_at <= now`. `engagement_gate` rows are created in one place: `waker::observe`, called from `POST /signals`. The only caller of `/signals` is the client's session-mutation layer, which emits `session-provisioned` when a session is **created**.

So the chain is: no session created ⇒ no signal ⇒ no gate row ⇒ never in `due` ⇒ never nudged. Not late. Never. The client's own module says so in as many words.

## Why the fix is not "insert a row whenever"

`waker::observe`'s comment contains the constraint that makes this delicate:

> No rows means never seen, and `from_storage` starts such a subject **full** rather than empty — an empty charge is instantly eligible, so the alternative would nudge a brand-new account before it did anything.

That is right and it must survive. Someone who installs the app at 9am must not be interrupted at 9:05. The row has to be created **full**, so `eligible_at` lands a few days out by the same arithmetic that governs everyone else — `Presence` has a three-day half-life and `THRESHOLD` is 110 of a possible 260, so a full subject who does nothing becomes eligible on the order of a week later. That is the intended behaviour, and it currently just never happens because the row is never written.

## Which event is "first contact"

The honest candidate is the **push subscription upsert**. `POST /push/subscribe` is a person explicitly saying "you may interrupt me" — the strongest possible statement of intent, and it happens exactly once per device before any study behaviour exists. `PushSubscriptionRepository` is already subject-keyed, so the id is right there.

Alternatives worth weighing in the PR, briefly:

- *Any authenticated request* — too broad; a page load is not consent.
- *A dedicated "hello" endpoint* — a new route to do what an existing one already implies.
- *Lazily, when the waker runs* — impossible; the waker can only see rows that exist. This is the circularity.

There is a real edge to name: a person who subscribes and then **unsubscribes every device** has a gate row and no way to be reached. `actuate` already handles zero accepted devices by releasing the claim, so this degrades correctly today — but it will `warn!` on a loop, and #253's log-noise sensibilities apply.

## Acceptance criteria

- [ ] A subject with no prior state gets an `engagement_gate` row on push subscription, seeded **full**, with `eligible_at` solved by the same arithmetic as everyone else.
- [ ] A test asserts a brand-new subject is **not** eligible immediately, and names the constraint from `observe`'s comment.
- [ ] A test asserts the same subject *does* become eligible after enough idle time, with the expected instant computed from `StudyCalibration` rather than hardcoded.
- [ ] Idempotent: re-subscribing, or subscribing a second device, does not reset or double an existing gate row.
- [ ] A subject with a gate row and zero live subscriptions does not produce repeated warnings on every pass.
- [ ] `docs/study-nudge.md` documents what creates a subject, since "nothing does" was the previous answer.

## Out of scope

- What is said once they are due. That is RCM2/RCM3 — landing this alone moves a subject from "never considered" to "considered, and correctly found to have nothing to say", which is progress and is testable on its own.
- Auth.

### PR DESCRIPTION
**Pull request by paulgsc**
## Description

`engagement_gate` rows — the waker's entire `WHERE eligible_at <= now` query surface — were created in exactly one place: `waker::observe`, called from `POST /signals`, whose only caller is the client's session-mutation layer, fired when a session is *created*. The chain was: no session ⇒ no signal ⇒ no gate row ⇒ never in `due` ⇒ never nudged. Not late — never. The client's own module said so in as many words. This is silence #1 in #257's cold-start argument.

`POST /push/subscriptions` closes it. `waker::first_contact` runs after the subscription itself is persisted: subscribing is a person explicitly saying "you may interrupt me" — the strongest statement of intent this deployment has, and it happens exactly once per device before any study behaviour exists. Alternatives considered and rejected, per the issue and recorded in `first_contact`'s doc comment:

- *Any authenticated request* — too broad; a page load is not consent.
- *A dedicated "hello" endpoint* — a new route restating what this one already implies.
- *Lazily, when the waker runs* — impossible; the waker can only see rows that already exist. This is the exact circularity the fix closes.

The row is seeded **full**, using the same `Charge::full`/`eligible_at` arithmetic `observe` already falls back to for an unseen subject — so a first-contact subject and one discovered through a stray signal land on identical footing, and `eligible_at` comes out on the order of a week later, not five minutes. Someone who installs the app at 9am must not be interrupted at 9:05.

`EngagementRepository::seed_if_absent` is new and deliberately **not** built on `save()`: `save`'s `ON CONFLICT` upserts both tables unconditionally, which is correct for folding in a new signal and wrong here — subscribing a second device, or re-subscribing the same one after key rotation, must not reset an already-drifting subject back to full. `INSERT OR IGNORE` against `engagement_gate`'s primary key is the guard, not a read-then-write, so two devices racing to be the first-ever subscription for a subject cannot both win.

The issue's named edge case — someone subscribes, then unsubscribes every device before ever becoming due — degrades correctly rather than looping loudly: `Presence` (fastest half-life, highest weight) stays the dominant deficit for a subject who has never received a single signal, so `GetStarted` (#294) keeps firing when they become due, and admission fails on `NotConsented` at `info!` rather than falling through to `Verdict::NothingToSay`'s `warn!`. Pinned by a new test in `waker.rs`.

`docs/study-nudge.md` gains a "First contact" section documenting what creates a subject, since "nothing does" was the previous answer.

Landing this closes **P0** — #295's first deployable increment — alongside #294 (P0.1, merged in #296): a fresh database, one subscription, no sessions, no signals, and the clock advanced is now enough to produce one notification that opens the app.

Fixes #278

Part of #295 (P0.2) and #257.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- `cargo check --workspace` and `cargo test --workspace` against a migrated SQLite database (via `sqlx-cli`, per `docs/study-nudge.md`) — all green, no new warnings under the workspace's `-D warnings` clippy pedantic/nursery/cargo lint set.
- `cargo sqlx prepare --workspace` succeeds, validating the new `INSERT OR IGNORE` queries in `EngagementRepository::seed_if_absent` against the real schema.
- `cargo fmt --all -- --check` clean on the changed files.
- New tests, one per acceptance-criterion checkbox that is unit-testable without a live DB connection in-process:
  - `study_domain::calibration::tests::first_contact_seeds_full_so_a_brand_new_subject_is_not_instantly_eligible` — a fresh full charge is not eligible at `now`, naming `observe`'s "starts full rather than empty" constraint.
  - `study_domain::calibration::tests::first_contact_does_become_eligible_once_the_full_charge_decays_to_threshold` — the same charge does cross the threshold, with the crossing instant taken from `Charge::eligible_at` rather than a hardcoded duration.
  - `file_host::nudge::waker::tests::a_first_contact_subject_who_unsubscribed_everywhere_is_suppressed_not_stuck_with_nothing_to_say` — the zero-live-subscriptions edge case resolves to `Verdict::Suppressed(NotConsented)`, not the `warn!`-logging `NothingToSay`.
  - Idempotency (`INSERT OR IGNORE` on `engagement_gate`'s primary key) and the "row created on first subscribe" behavior are structural/SQL-level guarantees, documented in `seed_if_absent`'s doc comment; the codebase has no precedent anywhere for a runtime DB-integration test harness (no `sqlx::test`/in-memory-pool usage exists in any of the 32+ prior PRs), so this follows the same repository-layer convention as `save`, `claim`, and `release`, none of which have direct tests either.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Additional Notes

Reference: #295 (the landing order) places this as **P0.2**, the second and last story in **P0**, the milestone's first deployable increment. #294 (P0.1, PR #296) is the other half.

---
_Generated by [Claude Code](https://claude.ai/code/session_01MNfgPbxx8BiY6daLjn9EYA)_

### CODE REVIEW

File: apps/servers/file_host/src/handlers/push.rs [hunk 1]
```diff
@@ -100,11 +100,21 @@ pub async fn subscribe(State(state): State<AppState>, subject: SubjectId, Json(r
...
 
+	// First contact: this is the event that says a subject exists at all —
+	// see `waker::first_contact` for why this route, and not a signal, is the
+	// honest place to seed the gate row. Propagated rather than swallowed:
+	// hiding this failure would silently recreate the exact "never due" bug
+	// (#278) this call exists to close, and the whole call is idempotent, so a
+	// client retry after a `500` is harmless.
+	crate::nudge::waker::first_contact(&state.core.shared_db, subject.as_str())
+		.await
+		.map_err(|err| FileHostError::OperationError(err.to_string()))?;
```
[the remark this answers is not in the document]
Comment (paulgsc) [reply apps/servers/file_host/src/handlers/push.rs:116]: Good catch — fixed in a895532. `waker::backfill_first_contact` now runs once at boot (alongside the waker spawn, gated the same way): it reads every distinct `subject_id` already in `push_subscriptions` via the new `PushSubscriptionRepository::distinct_subject_ids`, and calls `first_contact` for each. It's idempotent by construction (`first_contact` already is), so re-running it on every restart is a cheap no-op for anyone already gated, and one subject's failure is logged and skipped rather than aborting the pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
